packages/runtime/src/state/react-builder-preview.ts

```diff
@@ -402,15 +406,47 @@ function startPollingElementFromPoint(): ThunkAction<() => void, State, unknown,
   }
 }
 
-export function initialize(): ThunkAction<() => void, State, unknown, Action> {
-  return dispatch => {
+function registerBuilderComponents(): ThunkAction<() => void, State, unknown, Action> {
+  return (dispatch, getState) => {
+    const state = getState()
+    const componentsMeta = getComponentsMeta(state)
+
+    componentsMeta.forEach((meta, type) => {
+      const propControllerDescriptors = getComponentPropControllerDescriptors(state, type)
+      if (propControllerDescriptors != null) {
+        dispatch(registerComponent(type, meta, propControllerDescriptors))
+      }
+    })
+
+    return () => {
+      componentsMeta.forEach((_, componentType) => {
+        dispatch(unregisterComponent(componentType))
+      })
+    }
+  }
+}
```

Comment:
@migueloller Check this out, no channel needed here after all, we already handle `REGISTER_COMPONENT`/`UNREGISTER_COMPONENT` below, just needed to dispatch these two.

Reply:
Nice!

Reply:
Was there a difference between the `registerComponent` and the `registerBuilderComponent` actions? It seems that there would be given the naming, but perhaps it's silly tech debt.